File: Leetcode/PY/Leetcode447-回旋镖的数量.py
from typing import List
from collections import defaultdict


class Solution:
    def numberOfBoomerangs(self, points: List[List[int]]) -> int:
        res = 0
        for p in points:
            mp = defaultdict(int)
            for q in points:
                dis = (p[0] - q[0]) ** 2 + (p[1] - q[1]) ** 2
                mp[dis] += 1
            for _,v in mp.items():
                res += (v - 1) * v
        return res

File: Leetcode/sbgo/Leetcode447-回旋镖的数量.go
package sbgo

// 组合数

func numberOfBoomerangs(points [][]int) int {
	res := 0
	for _, p := range points {
		mp := make(map[int]int)
		for _, q := range points {
			dis := (p[0]-q[0])*(p[0]-q[0]) + (p[1]-q[1])*(p[1]-q[1])
			mp[dis]++
		}
		for _, v := range mp {
			res += (v - 1) * v
		}
	}
	return res
}
